### PR TITLE
Inserted unique id to the service name to avoid side effect

### PR DESCRIPTION
**Pull request by mangchiandjjoe**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -778,7 +778,7 @@ def generateServiceName(uniqueID, servicePrefix = '') {
   if (getParam('USE_LATEST_RELEASE', false) || getParam('GCLOUD_URL')) {
     return "${uniqueID}-dot-${PROJECT_ID}.appspot.com"
   }
-  return "${servicePrefix}testing-dot-${PROJECT_ID}.appspot.com"
+  return "${servicePrefix}-${uniqueID}-dot-${PROJECT_ID}.appspot.com"
```
Comment (qiwzhang) [inline Jenkinsfile]: This function is only used for GCE test.   GKE is here

https://github.com/cloudendpoints/esp/blob/master/script/e2e-kube.sh#L58

Comment (qiwzhang) [inline Jenkinsfile]: I think GCE tests are OK now.  The problem is in GKE

Comment (mangchiandjjoe) [inline Jenkinsfile]: Reverted https://github.com/cloudendpoints/esp/commit/a9a461d9c4ca021b9b7c36ce235dbd48f68bce46

File: Jenkinsfile [hunk 2]
```diff
@@ -761,12 +761,10 @@ def setupNode() {
...
 def getUniqueID(testId, useSha) {
   // This is used also for Kubernetes namespace and should not > 62 chars.
-  def date = new Date().format("yyMMddHHmm").toString()
-  def prefix = isReleaseQualification() ? 'rc-test' : 'test'
+  def prefix = isReleaseQualification() ? 'rc-' : ''
   def uuid = randomUUID().toString()
-  def sha = useSha ? "${GIT_SHA.take(7)}-" : ''
   def identifier = testId != '' ? "${testId}-" : ''
-  return "${prefix}-${sha}${identifier}${uuid.take(4)}-${date}"
+  return "${prefix}${identifier}${uuid.take(6)}"
```
Comment (qiwzhang) [inline Jenkinsfile]: Are we using a new service name for each test?   If so,  we may run out of 100 total services per project cap?   Do we delete the service for each run?

File: script/jenkins-utilities [hunk 3]
```diff
@@ -151,7 +151,7 @@ function create_service() {
     || error_exit 'Could not fetch the service version.'
 
   echo 'Activating service'
-  retry -n 3 run ${GCLOUD} service-management enable \
+  retry -n 10 run ${GCLOUD} service-management enable \
```
Comment (qiwzhang) [inline script/jenkins-utilities]: Do we need this command?

File: Jenkinsfile (hunk 2 above)
Comment (mangchiandjjoe) [inline Jenkinsfile]: The maximum length of the service name is 63. I had to remove some field to reduce the length.

e2e.sh and e2e-kube.sh are creating a new service and remove it in a cleanup function.
Currently the e2e test creates and removes 12 services. 

When both e2e and pr-e2e jobs are running concurrently, then we could use maximum 24 services.

For now endpoints-jenkins project has 53 services which is not managed by e2e tests.
We need to remove services not in use.

https://pantheon.corp.google.com/endpoints?project=endpoints-jenkins&organizationId=433637338589

File: script/jenkins-utilities (hunk 3 above)
Comment (mangchiandjjoe) [inline script/jenkins-utilities]: gcloud service-management deploy command will activate the service.
I think we can remove this.

I saw a case this command was failed. So I increased the number of retry.

File: Jenkinsfile (hunk 2 above)
Comment (qiwzhang) [inline Jenkinsfile]: My concern is: if test is aborted, service may not get cleaned properly.  we will have some service leaking.   If 2, or 3 such leaking,  we will run out of 100.

Can we not random the service name?   each test use its own service name,  just make sure they are not sharing name.   same e2e test may run in different stage: e2e and release-qualify, make sure they are not using the same service name too

**Pull request closed without merging**